ccl/changefeedccl/tableset: TestTablesetBasic failed

ccl/changefeedccl/tableset.TestTablesetBasic [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20531829?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20531829?buildTab=artifacts#/) on master @ [892744fafd62817c45ff95f4d56216c12fb7a5d1](https://github.com/cockroachdb/cockroach/commits/892744fafd62817c45ff95f4d56216c12fb7a5d1):


```
=== RUN   TestTablesetBasic
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/a1855337a6eb59e75bdd00ecc0e5565d/logTestTablesetBasic2161632680
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
IsMetamorphicBuild==false
=== RUN   TestTablesetBasic/filter=Filter{database_id=100,_include_tables=[foober_foo_bar_baz],_exclude_tables=[]}
=== RUN   TestTablesetBasic/filter=Filter{database_id=100,_include_tables=[foober_foo_bar_baz],_exclude_tables=[]}/drop_watched_table
panic: Counters should not decrease, prev: 2752823296, new: 0.

goroutine 878 [running]:
github.com/cockroachdb/cockroach/pkg/util/metric.(*Counter).Update(0xc001b33d60, 0x0)
	pkg/util/metric/metric.go:829 +0x106
github.com/cockroachdb/cockroach/pkg/server/status.(*RuntimeStatSampler).SampleEnvironment(0xc001b8d208, {0x86dd6bd0, 0xc001a4e460}, 0xc00510c950)
	runtime.go:1047 +0x71a
github.com/cockroachdb/cockroach/pkg/server.startSampleEnvironment.func1({0x86dd6bd0, 0xc001a4e460})
	pkg/server/env_sampler.go:186 +0x1a6
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x86dd6bd0, 0xc001a4e460})
	pkg/util/stop/stopper.go:440 +0xac
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 50
	pkg/util/stop/stopper.go:438 +0x16c
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTablesetBasic.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54897

--- a/pkg/util/metric/metric.go
+++ b/pkg/util/metric/metric.go
@@ -825,7 +825,7 @@ func (c *Counter) Inc(v int64) {
 // maintained elsewhere.
 func (c *Counter) Update(val int64) {
 	if buildutil.CrdbTestBuild {
-		if prev := c.count.Load(); val < prev {
+		if prev := c.count.Load(); val < prev && val != 0 {
 			panic(fmt.Sprintf("Counters should not decrease, prev: %d, new: %d.", prev, val))
 		}
 	}
@@ -1446,7 +1446,7 @@ func (cv *CounterVec) Update(labels map[string]string, v int64) {
 	}
 
 	currentValue := cv.Count(labels)
-	if currentValue > v {
+	if currentValue > v && v != 0 {
 		panic(fmt.Sprintf("Counters should not decrease, prev: %d, new: %d.", currentValue, v))
 	}
 

